sdk/keyvault/keyvault-keys/src/lro/delete/poller.ts

```diff
@@ -0,0 +1,55 @@
+import { delay, RequestOptionsBase } from "@azure/core-http";
+import { Poller } from "@azure/core-lro";
+import { DeleteKeyPollOperationState, makeDeleteKeyPollOperation } from "./operation";
+import { DeletedKey, KeyClientInterface } from "../../keysModels";
+
+export interface DeleteKeyPollerOptions {
+  client: KeyClientInterface;
+  name: string;
+  requestOptions?: RequestOptionsBase;
+  intervalInMs?: number;
+  resumeFrom?: string;
+}
+
+/**
+ * Class that deletes a poller that waits until a key finishes being deleted
+ */
+export class DeleteKeyPoller extends Poller<DeleteKeyPollOperationState, DeletedKey> {
+  /**
+   * Defines how much time the poller is going to wait before making a new request to the service.
+   * @memberof DeleteKeyPoller
+   */
+  public intervalInMs: number;
+
+  constructor(options: DeleteKeyPollerOptions) {
+    const { client, name, requestOptions, intervalInMs = 1000, resumeFrom } = options;
```

Comment:
I feel like the default interval should be more than a second, but maybe that's just me.

Reply:
How much should it be?

Reply:
I think 1 second is not horrible for this particular case, but anywhere between 1-5 seconds would be ok.